tensorflow/core/common_runtime/mkl_cpu_allocator.h

```diff
@@ -48,6 +50,120 @@ class MklSubAllocator : public SubAllocator {
   void Free(void* ptr, size_t num_bytes) override { port::AlignedFree(ptr); }
 };
 
+/// CPU allocator that handles small-size allocations by calling
+/// suballocator directly. Mostly, it is just a wrapper around a suballocator
+/// (that calls malloc and free directly) with support for bookkeeping.
+class MklSmallSizeAllocator : public VisitableAllocator {
+ public:
+  MklSmallSizeAllocator(SubAllocator* sub_allocator, size_t total_memory,
+                        const string& name) : sub_allocator_(sub_allocator),
+                        name_(name) {
+    stats_.bytes_limit = total_memory;
+  }
+  ~MklSmallSizeAllocator() override {}
```

Comment:
Hi @penpornk since MklSmallSizeAllocator is just a wrapper over libc malloc/free, it does not need to deallocate anything in destructor. The constructor is not allocating any memory either. Makes sense?

Reply:
I was talking about the memory regions allocated through AllocateRaw. It's possible the user forget to free some of them through DeallocateRaw. Since libc don't keep track of (and free) its allocations, shouldn't we check if there is any region left and free them here?

Reply:
I think my question is, do we expect the allocated memory to last longer than the allocator?

Reply:
I think that is considered a memory leak, and an allocator would not be responsible for it, right? A user needs to ensure that the memory is released.

Reply:
True. Let's leave it like this then.